### ISSUE TITLE
R7xx

### ISSUE THREAD

**Issue by emoose** [emoose/re4-research#18]
Made a comment on re4hd mentioning how R7xx could be connected to EM_LIST_BIO5 found in the PS2 symbols:
https://www.re4hd.com/?p=9649#comment-202588

Also made another comment there but not sure if it went through, maybe blog software/anti-spam swallowed it up:

---

In the executable there's a list of strings associated with areas in the game:
![image](https://i.imgur.com/SPrEQ5K.png)

biox is at index 9 into the list, the game also has a function "checkEmListNo" which takes a room number (eg R703) and returns the index for it in this area list.

Anyway, for room 6xx & 7xx:
![image](https://i.imgur.com/Inf4t2y.png)

So **6xx falls under 18 ps2-tgs, and 7xx is under 9 biox**.

However, if we look at debug symbols in the recent debug PS2 release, there's an interesting enum inside it:

```
typedef enum {
   EM_LIST_FALSE = -1,
   EM_LIST_ST1_0, 
   [...]
   EM_LIST_BIO5 = 9, 
   EM_LIST_ADA, 
   EM_LIST_ETC, 
   EM_LIST_ETC2, 
   EM_LIST_PS2ST1, 
   EM_LIST_PS2ST2, 
   EM_LIST_PS2ST3, 
   EM_LIST_PS2ST4, 
   EM_LIST_PS2ST5, 
   EM_LIST_PS2ETC
} EM_LIST_KIND;
```

The entries in this enum match up with the strings in the game executable, though in this enum biox is named BIO5 instead 🫨
(most likely devs knew the bio5 string would end up in the exe, and since this was long before RE5 released, they changed it to biox to avoid leaking it too early, but the actual enum in the code was left alone)

So what does this all mean? **The R7XX maps fall under BIO5**!

Looks like RE5 was being prototyped on the RE4 engine during the time between GC & PS2 release, and guess the files for that prototype ended up being left inside the RE4 source tree (original devs probably knew they could remove them before shipping the final PS2/Wii builds, but luckily for us X360/PS3 was done by a new team 😄)

---

Also found out that EM_LIST_BIO5 is tied to the emleon09.esl file, on GC/UHD that file seems to just contain enemies for some R1XX rooms, but the PS2 version of emleon09.esl actually seems to have enemies linked to both R703 & R702! Might be worth exploring some more once we can load in the R7XX maps properly.

**Comment by emoose** [comment]
Yep, looks like the PS2 emleon09 does go with those maps :) Spawns like 30 enemies on the ground, with some also on the roofs as seen here:

![20220818161954_1](https://user-images.githubusercontent.com/1755499/185432419-7dd725a1-ad9c-4185-8b5c-e360e2164680.jpg)

For that copy st0.rel as st7_0.rel, remove emleon09.esl.lfs, and copy in emleon09.esl from PS2 version in its place, ~~then use CE and set (byte, 1.1.0) `bio4.exe+813CD0+18` to 1 (maybe backup your savegame first though), and then area-jump to R702 (edit `AREAJUMP_MAX_STAGE` to 7 inside UI_DebugWindows.cpp first)~~ (shouldn't be needed with latest re4_tweaks trainer build)

That should let you load into R702 fine, sadly R703 is currently crashing inside `calcModelAddr`, something to do with `Smd::setObj` called by `gameRoomInit`, but that does seem to have some different Em ID spawned inside it.

E: looks like R703 crash is caused by first 32-bits inside the SMD contained in R703.udas, endian-swapping it seems to let game load in, but level is all black with no geometry, not sure if that's something caused by the SMD edit, or maybe the level just sucks...
(same for the R700/R701 files too, but R703 did load some enemies from the ESL at least)

---

R6xx maps seem to load in fine with `St6_data_tbl` save_flg set, so hopefully adding a dummy buffer like PS2 would probably help fix all those too.

**Comment by Atomoksetin1996** [comment]
The emleom09.esl file is also present in the gamecube version, in which there were no St5,6,7 yet. Judging by the IDs of the enemies and their subtypes, we can conclude that this file was saved in a very early build of RE 4.
This file was last modified no later than the American trial version.
For example, the value 15 01 spawns a ganado in a blue shirt,
already in the Japanese trial version, it was replaced by a female ganado with the identifier 15 0B, so the value 15 01 does not exist in the final version. The same can be said about other enemy IDs, for example, 16 00 in this file is not a ganado, but another unknown enemy type.
In earlier versions of RE 4, each individual ganado had its own unique ID. Ganados were later grouped into groups within the same archive with the same ID, but sometimes old enemy IDs are mentioned in the game.

Let's say the textures and folders of opponents in the PS2 version are numbered according to the old enemy numbering.
